### PR TITLE
remove CSI-migration gate

### PR DESCRIPTION
**Pull request by cyclinder**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

As a part of https://github.com/kubernetes/kubernetes/pull/114068, We do CSI-migration gate removals in this PR.

#### Does this PR introduce a user-facing change?

```release-note
The feature gates `CSIMigrationGCE` is graduated to GA and were unconditionally enabled have been removed in v1.25, and the entire gcepd package has been removed. 
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by haosdent** [comment]
> Which issue(s) this PR fixes:
> Fixes #

Need to link to the correct issue? @cyclinder

**Comment by cyclinder** [comment]
> Need to link to the correct issue?

This PR does not have an issue to fix. Just to complete the cleanup of `csi-migration` gate, the original request originated from https://github.com/kubernetes/kubernetes/pull/114068#issuecomment-1482914741

**Comment by haosdent** [comment]
Got it, @cyclinder

**Comment by haosdent** [comment]
LGTM

**Comment by liggitt** [comment]
/unassign
this should get acked by storage approvers first
/assign @msau42

File: cmd/kube-controller-manager/app/plugins_providers.go
Comment (msau42) [inline cmd/kube-controller-manager/app/plugins_providers.go:30]: Also remove the entire gcepd package here: https://github.com/kubernetes/kubernetes/tree/master/pkg/volume/gcepd

File: pkg/features/kube_features.go
Comment (msau42) [inline pkg/features/kube_features.go:172]: I would put CSIStorageCapacity removal in a different PR and make this one only about gcepd.

**Comment by msau42** [comment]
cc @mattcary

File: cmd/kube-controller-manager/app/plugins_providers.go
Comment (mattcary) [reply cmd/kube-controller-manager/app/plugins_providers.go:30]: There are several other gcepd references that would have to be cleaned up first:

```
pkg/volume/gcepd/doc.go:19:package gcepd // import "k8s.io/kubernetes/pkg/volume/gcepd"
pkg/volume/util/operationexecutor/operation_generator_test.go:42:	"k8s.io/kubernetes/pkg/volume/gcepd"
pkg/kubelet/kubelet_test.go:95:	"k8s.io/kubernetes/pkg/volume/gcepd"
```

Comment (cyclinder) [reply cmd/kube-controller-manager/app/plugins_providers.go:30]:
```
➜  kubernetes git:(feature_gate) grep 'k8s.io/kubernetes/pkg/volume/gcepd' -r .           
./cmd/kube-controller-manager/app/plugins_providers.go: "k8s.io/kubernetes/pkg/volume/gcepd"
./cmd/kubelet/app/plugins_providers.go: "k8s.io/kubernetes/pkg/volume/gcepd"
./pkg/kubelet/kubelet_test.go:  "k8s.io/kubernetes/pkg/volume/gcepd"
./pkg/volume/util/operationexecutor/operation_generator_test.go:        "k8s.io/kubernetes/pkg/volume/gcepd"
./pkg/volume/gcepd/doc.go:package gcepd // import "k8s.io/kubernetes/pkg/volume/gcepd"
```

I'm not sure the entire gcepd package should be removed, `CSIMigrationGCE` is GA in 1.25 and still using it, Typically we wait for two releases to remove the feature gate, refer to commit: https://github.com/kubernetes/kubernetes/pull/114068#discussion_r1105456171 . So Is this the time to remove?

https://github.com/kubernetes/kubernetes/blob/ad18954259eae3db51bac2274ed4ca7304b923c4/cmd/kube-controller-manager/app/plugins_providers.go#L65

https://github.com/kubernetes/kubernetes/blob/ad18954259eae3db51bac2274ed4ca7304b923c4/pkg/features/kube_features.go#L135

File: pkg/features/kube_features.go
Comment (cyclinder) [reply pkg/features/kube_features.go:172]: How about putting it in https://github.com/kubernetes/kubernetes/pull/114068 ?

File: cmd/kube-controller-manager/app/plugins_providers.go
Comment (msau42) [reply cmd/kube-controller-manager/app/plugins_providers.go:30]: This is targeting 1.28 which is more than 2 releases since GA.

Comment (cyclinder) [reply cmd/kube-controller-manager/app/plugins_providers.go:30]: Updated.

File: pkg/features/kube_features.go
Comment (xing-yang) [reply pkg/features/kube_features.go:179]: Looks like #114068 is already merged.
There's another PR submitted to remove CSIStorageCapacity feature gate: https://github.com/kubernetes/kubernetes/pull/118018
The title for this PR is "remove CSI-migration gate" so it's better to only remove CSI-migration related things.

Comment (cyclinder) [reply pkg/features/kube_features.go:179]: yes, this PR only removes CSI-migration related things.

**Review by humblec — CHANGES_REQUESTED** [review]
The test case changes have to be revisited.

File: pkg/volume/util/operationexecutor/operation_generator_test.go
Comment (cyclinder) [reply pkg/volume/util/operationexecutor/operation_generator_test.go:61]: Thanks for the review @humblec.  the entire gcepd package has been removed. so the pluginName: GCEPDInTreePluginName must be changed. Or remove this test case. HDYT?

Comment (msau42) [reply pkg/volume/util/operationexecutor/operation_generator_test.go:61]: Try another plugin that can be disabled, like portworx

Comment (cyclinder) [reply pkg/volume/util/operationexecutor/operation_generator_test.go]: Thanks @msau42 , Now updated to `PortworxVolumePluginName`.

Comment (cyclinder) [reply pkg/volume/util/operationexecutor/operation_generator_test.go]: I'm not familiar with CSI's code, I saw the git history and I think we should remove this case in the test file.

Comment (cyclinder) [reply pkg/volume/util/operationexecutor/operation_generator_test.go]: Or using `fake-plugin`, just like https://github.com/kubernetes/kubernetes/pull/117055/commits/81b4cfb9b3d43bd522d63bb067e13a605167d6ad

**Comment by msau42** [comment]
Thanks this is looking good. Can you update the release note now that only one feature gate is being removed?

**Comment by cyclinder** [comment]
> Thanks this is looking good. Can you update the release note now that only one feature gate is being removed?

Sure, updated now

**Comment by cyclinder** [comment]
PR has been ready, Please take a review, Thanks.

/cc @msau42 @xing-yang @humblec

**Comment by msau42** [comment]
@cyclinder sorry the delay. Can you update remove the references to Azure feature gates from the release note? It looks like this PR is only modifying GCE PD.

/lgtm
/approve
/hold

to fix release note

**Comment by cyclinder** [comment]
thanks @msau42 ❤️

**Pull request merged**
